### DISCUSSION TITLE
Hivemq cloud connection

### DISCUSSION THREAD

**Discussion by ivangilli** [jziolkowski/tdm#247] [Ideas]
Is it possible to connect with the hivemq cloud mqtt service? Currently not...

**Comment by barbudor** [comment]
Yes it is
You need a build with mqtt-tls enabled which is not by default for esp8266 but is for esp32

This has been addressed multiple times in discussions in the past, please search

**Reply by ivangilli** [reply]
??…the question is tdmgr related…not for tasmota…I know how to connect tasmota to mqtt with tls support

**Reply by barbudor** [reply]
Sorry my bad. I didn't noticed this was on TDM, not Tasmota 🤦

AFAIK for now TDM doesn't support TLS
Still in the to-do list

**Reply by barbudor** [reply]
Hum, reading back some pas messages, it seems that it should be available in dev version

**Reply by ivangilli** [reply]
Yes, but how to implement?

**Reply by barbudor** [reply]
Just took me 5 minutes to figure out

Here is a configuration that works for me:
![image](https://github.com/jziolkowski/tdm/assets/2996042/9ad0256d-70cf-45bc-b109-c9956742c1cd)

Hostname should be the URL of the cluster you created by creating an account at HiveMQ
You also create a client credentials. Here I have created one with username "barbudor" and a password

It is important to check the "TLS Insecure" box which will avoid validating the TLS certificate of HiveMQ, just perforing encryption.
Otherwise you need to have a certificate in /etc/opentl/certs/ca.crt (or somewhere else)

You can also optionally check the box "connect on startup"

Hope this helps

**Reply by barbudor** [reply]
Actually you don't need to create a username/password
Leaving that blank also works fine

Indeed the cluster URL is a token which is private to your account.
Creating client credentials would allow, for example, to create access rules to topics based on client

**Reply by ivangilli** [reply]
How can I show this window with optional features??

**Reply by barbudor** [reply]
This is the standard MQTT Broker configuration for the current development version
You must pull the development branch from GitHub and run it in Python.
TLS is (AFAIK) not available in binary release

@jziolkowski @Jason2866 Would it be possible to update PyPi and binary release to the latest development version ?

Thanks

**Reply by ivangilli** [reply]
I run on Mac but when try to connect at broker tdmgr crash… :/ openssl or cert problem…I will make cert file and retry

**Reply by ivangilli** [reply]
On windows works with out issue thanks!!

**Comment by ivangilli** [comment]
On Mac there is a problem:
2023-07-29 14:51:49 [INFO] ### TDM START ###
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
QWidget::insertAction: Attempt to insert null action
qt.qpa.window: <QNSWindow: 0x7fde4f2f6000; contentView=<QNSView: 0x7fde4f2f5bf0; QCocoaWindow(0x600002f80210, window=QWidgetWindow(0x600003ca6100, name="DevicesListWidgetClassWindow"))>> has active key-value observers (KVO)! These will stop working now that the window is recreated, and will result in exceptions when the observers are removed. Break in QCocoaWindow::recreateWindowIfNeeded to debug.
Traceback (most recent call last):
  File "/Users/ivan/Downloads/tdm-develop/tdmgr.py", line 225, in toggle_connect
    self.mqtt_connect()
  File "/Users/ivan/Downloads/tdm-develop/tdmgr.py", line 258, in mqtt_connect
    self.mqtt.connectToHost()
  File "/Users/ivan/Downloads/tdm-develop/Util/mqtt.py", line 142, in connectToHost
    self.m_client.connect(self.m_hostname, port=self.port, keepalive=self.keepAlive)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/paho/mqtt/client.py", line 1073, in reconnect
    sock.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 1346, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
zsh: abort      python3 /Users/ivan/Downloads/tdm-develop/tdmgr.py


Where is the configura file on Mac?

**Reply by ivangilli** [reply]
I was able to make the program work on Mac too...I had to edit the mqtt.py management file
